### DISCUSSION TITLE
Dynamic Layout on the same page?

### DISCUSSION THREAD

**Discussion by indicozy** [vercel/next.js#41644] [Help] [answered]
Hi everyone!

I'm trying to build layout system with different types of layout for Learning Management System. For instance, if this page should render Quiz block, it should only provide static page of Quiz. If it is Code Assessment, it should provide IDE. How can I do that so the user will not get 3 different layout which will be rendered on the client side? Can NextJS handle rendering different layouts on SSR or SSG?

Example:
```jsx
export default function Elements({type}) {
if (type === 1) return (<Layout1/>)  
if (type === 2) return (<Layout2/>) 
if (type === 3) return (<Layout3/>) 
}
export function async getStaticProps(){
await dbInfo = await getLessonInfo()
return {type: dbinfo.type}
}
```

**Comment by leerob** [comment, marked as answer]
**Update:** We’re excited to announce [Next.js 13](https://nextjs.org/13), which includes the new `app/` directory (beta) with support for nested layouts, colocation of data fetching, streaming, Server Components by default, and much more.

Inside the `app/` directory, there is a powerful new way to fetch data with React’s `use()` hook and the extended Web `fetch()` API. This allows you to colocate data fetching directly inside components for the most flexibility, including inside layouts.

You can leave feedback here: https://github.com/vercel/next.js/discussions/41745
